lib/event_stream.rb

```diff
@@ -38,24 +39,39 @@ def publish(name, attrs = {})
     # Or, any arbitrary predicate on events may be provided.
     # @yield [Event] action to perform when the event occurs.
     def subscribe(filter = nil, &action)
-      filter ||= lambda { |e| true }
-      filter_predicate = case filter
-                         when Symbol, String then lambda { |e| e.name.to_s == filter.to_s }
-                         when Regexp then lambda { |e| e.name =~ filter }
-                         when Hash then lambda { |e| filter.all? { |k,v| e[k] === v } }
-                         else filter
-                         end
-      @subscribers << Subscriber.new(filter_predicate, action)
+      add_subscriber(Subscriber.create(filter, &action))
     end
 
     # Clears all subscribers from this event stream.
     def clear_subscribers
       @subscribers = []
     end
+
+    # Returns all subscribers for this stream
+    # @return [Array<EventStream::Subscriber]
+    def subscribers
+      @subscribers
+    end
+
+    # Adds a subscriber to this stream
+    # @param [EventStream::Subscriber]
+    def add_subscriber(subscriber)
+      @subscribers << subscriber
+    end
   end
 
-  # @private
   class Subscriber < Struct.new(:filter, :action)
+    def self.create(filter = nil, &action)
+      filter ||= lambda { |e| true }
+      filter_predicate = case filter
```

Comment:
I think this case statement should be documented as it seems like a critical part of the API